File: binary_search.js
const binarySearch = (array, search) => {
    if(array.length < 1) {
        return false;
    }
    const length = array.length;
    const midIdx = Math.floor(length/2);
    const mid = array[midIdx];
    const left = array.slice(0, midIdx);
    const right = array.slice(midIdx + 1);

    if(mid === search) {
        return midIdx;
    } else if(mid > search) {
        return binarySearch(left, search);
    } else {
        res = binarySearch(right, search);
        if(res !== false) {
            return res + midIdx + 1;
        } else {
            return res;
        }
    }
}

File: binary_search.py
def binary_search(sorted_list, search):
    length = len(sorted_list)
    if length < 1:
        return False
    mid_index = int(length/2)
    mid_value = sorted_list[mid_index]
    right = sorted_list[mid_index + 1:]
    left = sorted_list[:mid_index]
    if mid_value == search:
        return mid_index
    elif mid_value > search:
        return binary_search(left, search)
    else:
        result = binary_search(right, search)
        if result != False:
            return result + mid_index + 1
        else:
            return result
